### ISSUE TITLE
large chromosome simulation,multithread?

### ISSUE THREAD

**Issue by cistarsa** [tskit-dev/msprime#2024]
Hello, I'm attempting to generate a neutral vcf for 40 individuals on a 50Mb chromosome using the command line as ```msp ancestry``` and ```msp mutation``` and was wondering if I can multi-thread this process on a slurm cluster or a high performance node with 96 cores. As of now this is only using 1 CPU and 0.5% memory, thank you.

**Comment by molpopgen** [comment]
Hi @cistarsa --

Welcome, and thanks for the question.  These simulation algorithms do not support parallel computing.  Would you mind sharing your command line for `msp ancestry`?  The way that recombination rate interacts with some other parameters can have a big impact on performance.

**Comment by cistarsa** [comment]
Ah sure, yeah so even on a high performance node with many cpus it only uses 1 core at a time? Is there a way to have it take full advantage of the system it's on? Thanks and here's what I've been using

```
msp ancestry 25 -N 40071 -L 5774900 --recombination-rate 2.95e-6 --ploidy 2 > LI_25.trees
```

**Comment by molpopgen** [comment]
> Ah sure, yeah so even on a high performance node with many cpus it only uses 1 core at a time? Is there a way to have it take full advantage of the system it's on? Thanks and here's what I've been using
> 
> ```
> msp ancestry 25 -N 40071 -L 5774900 --recombination-rate 2.95e-6 --ploidy 2 > LI_25.trees
> ```

No, parallel methods can only be used if the algorithms can be broken into approximately independent bits. For coalescent models, it is not clear that doing so is possible.

**Comment by molpopgen** [comment]
One question: where does that recombination rate come from?:. It is an order or magnitude or two higher than typical values used for, e.g., humans.

**Comment by cistarsa** [comment]
Gotcha, I see the same single-threading with Slim, too... hmm and yes,
typical rate for some taxa but this was estimated using pyrho
<https://github.com/popgenmethods/pyrho#compute_r2> for insect, which is on
par.

**Comment by jeromekelleher** [comment]
Hi @cistarsa :wave: As @molpopgen says it's not trivial to break up these simulation algorithms into parallel chunks. The main reason that we haven't put any effort into this is that we're nearly always running many replicates when doing such simulations, and these parallelise naturally.
